gcp-deployer/cmd/create.go

```diff
@@ -69,14 +71,28 @@ func RunCreate(co *CreateOptions) error {
 		return err
 	}
 
+	ca, err := loadCAOrNil()
+	if err != nil {
+		return err
+	}
+
 	d := deploy.NewDeployer(provider, kubeConfig, co.MachineSetup)
 
-	return d.CreateCluster(cluster, machines)
+	return d.CreateCluster(cluster, machines, ca)
 }
+
 func init() {
 	createCmd.Flags().StringVarP(&co.Cluster, "cluster", "c", "", "cluster yaml file")
 	createCmd.Flags().StringVarP(&co.Machine, "machines", "m", "", "machine yaml file")
 	createCmd.Flags().StringVarP(&co.MachineSetup, "machinesetup", "s", "machine_setup_configs.yaml", "machine setup configs yaml file")
+	createCmd.Flags().StringVarP(&co.CertificateAuthorityPath, "certificate-authority-path", "a", "", "location of a custom certificate authority to be used on a new cluster, if not provided one is generated")
```

Comment:
Could we have more details on what is expected by the code at that path? For example, the code seems to expect some file name conventions. Hopefully, the user of this tool does not need to read the code to figure out what is needed.

Reply:
Agree, this help message needs to be more explanatory for the user -- will update.

Reply:
Updated with the following help message:

```
  -a, --certificate-authority-path string   optional path to a custom certificate authority to be used on a new cluster, path can be one of the following:
                                             1. directory: a path to a directory, the directory must contain two files named ca.crt and ca.key containing the certificate and private key respectively.
                                             2. certificate: a path to a certificate file, ${filename}.crt, the file must end with extension '.crt' and there must be a file named ${filename}.key in the same directory."
                                             3. key: a path to a key file, ${filename}.key, the file must end with extension '.key' and there must be a file named ${filename}.crt in the same directory.
```